cryptoservice/crypto_service.go

```diff
@@ -110,28 +100,23 @@ func (ccs *CryptoService) Sign(keyIDs []string, payload []byte) ([]data.Signatur
 			}
 		}
 
-		algorithm := privKey.Algorithm()
 		var sigAlgorithm data.SigAlgorithm
-		var sig []byte
 
-		switch algorithm {
-		case data.RSAKey:
-			sig, err = rsaSign(privKey, payload)
+		switch privKey.(type) {
+		case *data.RSAPrivateKey, *data.RSAx509PrivateKey:
```

Comment:
In this context it means it's an RSA private key with an x509 Public component. Have a look at the actual key definitions in tuf/data/keys.go to see the difference between RSAPrivateKey and RSAx509PrivateKey